digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/AssetDataImporter.scala

```diff
@@ -858,4 +858,201 @@ class AssetDataImporter {
+              {
+                var sweAddressAdded=false;
+                finstops.foreach{finstop=> //finstop  _1:asset_id, _2:address,_3:link-id
+                {
+                if (finstop._3==stops._2)
+                  {
+                    val swedishaddress=rlinks.attributes.getOrElse("ROADNAME_SE","none").toString
+                    if (swedishaddress!="none" && swedishaddress!=null && rlinks.attributes.getOrElse("ROADNAME_FI","null").toString.toUpperCase()==finstop._2.toUpperCase())
+                     {
+                       createTextPropertyValue(stops._1, 300096, swedishaddress)
+                       sweAddressAdded=true;
+                     }
+                  }
+                }}
+                  if (!sweAddressAdded)
+                  swedishstops.foreach{swestop=> //swestop  _1:asset_id, _2:address,_3:link-id
+                    if (swestop._3==rlinks.linkId)
+                    {
+                      val finAddress=rlinks.attributes.getOrElse("ROADNAME_FI","none").toString
+                      if (finAddress!="none" && finAddress!=null && rlinks.attributes.getOrElse("ROADNAME_SE","null").toString.toUpperCase()==swestop._2.toUpperCase())
+                      {
+                        createTextPropertyValue(stops._1, 300095, finAddress)
+                      }
+                    }
+
+                  }
+                }
+              }
+          }
+        }
+      }
+    }
+
+    /**
+      * Gets municipalitycodes of stops which have updateable masstransitstops
+      * returns list of int
+      */
+    def getMTStopsMunicipalitycodeBothMissing() =
+    {
+
+      sql"""
+              Select distinct MUNICIPALITY_CODE
+                            From Asset
+                   WHERE
+                   Asset_Type_ID=10
+                    AND
+                    (
+                   (ID NOT IN (SELECT ASSET_ID FROM Text_property_value WHERE PROPERTY_ID=300096 OR PROPERTY_ID=300095 ))
+                    )
+               ORDER BY MUNICIPALITY_CODE DESC""".as[(Int)].list
+    }
+
+  /**
+    * Retrives distinct int list filled with  municipalitycodes of mass transit stops which have only finnish OR swedish address
+    * @return
+    */
+
+  def getMTStopsMunicipalitycodeOneMissing() =
+  {
+
+    sql"""
+                    Select distinct a.MUNICIPALITY_CODE
+                    From Asset a, Text_property_value v, Text_property_value fiv
+                    WHERE
+                     a.Asset_Type_ID=10
+                      AND
+                      ((fiv.PROPERTY_ID =300096  AND   (fiv.Asset_ID NOT IN (Select Asset_ID From Text_property_value Where PROPERTY_ID=300095)  AND a.ID=fiv.ASSET_ID))
+                      OR
+                      (fiv.PROPERTY_ID =300095  AND   (fiv.Asset_ID NOT IN (Select Asset_ID From Text_property_value Where PROPERTY_ID=300096)  AND a.ID=fiv.ASSET_ID)))
+               ORDER BY MUNICIPALITY_CODE DESC""".as[(Int)].list
+  }
```

Comment:
not sure i understood comment correctly... You mean i should   for example use Asset_type table to determine which asset type corresponds to mass transit stop instead of number 10?